### PR TITLE
Feature navigation

### PR DESCRIPTION
**Pull request by junaidrsd**
Side navigation
Third navigation level
Prevent to load sections on main navigation level
Hamburger menu styles on mobile view (only with sidebar nav)

### CODE REVIEW

**Comment by MatthieuScarset** [comment]
Seems to include great enhancements!
Hopefully this pull make it throught acceptation process.
Thanks @junaidrsd

File: gulpfile.babel.js [hunk 1]
```diff
@@ -80,6 +80,7 @@ gulp.task('dev:generate', () => {
   return gulp.src(['lib/app/css/*.css'])
     .pipe(styleguide.generate({
       title: 'SC5 Styleguide',
+      sideNav: true,
```
Comment (ramert) [inline gulpfile.babel.js]: Will this be the new default value, so if you update toNav changes to sideNav? If so, I would change default value to false.

File: lib/app/views/main.html [hunk 2]
```diff
@@ -10,47 +10,109 @@ <h1 ng-if="status.hasError">Error: {{status.error.name}}</h1>
...
 </header>
-<nav class="sg sg-top-nav" ng-class="{'designerToolVisible' : designerTool.isVisible}">
+
+<nav class="sg sg-top-nav" ng-class="{'designerToolVisible' : designerTool.isVisible}" ng-if="!config.data.sideNav">
   <ul class="sg sg-top-nav-menu">
     <li>
       <a class="sg"
         ng-class="currentSection === 'overview' ? 'active' : ''"
-        ng-click="clearSearch()"
+        ng-click="clearSearch();"
```
Comment (ramert) [inline lib/app/views/main.html]: If only one function call is used in ng-events, good practise is not to use semi-colon.

File: lib/app/css/styleguide-app.css [hunk 3]
```diff
@@ -344,40 +345,255 @@ markup:
...
+    display: inline-block;
+    width: 30%;
+    opacity: 1;
+    @media (--mobile) {
+      width: 100%;
+      padding-top: 30px;
+      z-index: 99999;
+      background: #8b8b8b;
+      overflow: scroll;
+      transition: opacity 400ms cubic-bezier(0.000, 0.995, 0.990, 1.000);
```
Comment (ramert) [inline lib/app/css/styleguide-app.css]: Two different notations for time: 400ms and .4s. Perhaps, only either is enough?

File: lib/app/css/styleguide-app.css [hunk 4]
```diff
@@ -344,40 +345,255 @@ markup:
...
+    & .sg-nav-section li {
+      margin-bottom: .3em;
+      @media (--mobile) {
+        margin-bottom: 0;
+      }
+    }
+
+    & .sg-nav-section li a {
+      @media (--mobile) {
+        color: white;
```
Comment (ramert) [inline lib/app/css/styleguide-app.css]: Should this color be parametrized?

**Pull request merged**